include/deal.II/lac/constraint_matrix.h

```diff
@@ -935,9 +943,9 @@ class ConstraintMatrix : public Subscriptor
    * eigenvalue (with a multiplicity that is possibly greater than one).
    * Taking this into account, nothing else has to be changed.
    */
-  template <typename MatrixType>
+  template <typename MatrixType, typename LocalType>
   void
-  distribute_local_to_global (const FullMatrix<double>     &local_matrix,
+  distribute_local_to_global (const FullMatrix<LocalType>  &local_matrix,
                               const std::vector<size_type> &row_indices,
                               const std::vector<size_type> &col_indices,
                               MatrixType                   &global_matrix) const;
```

Comment:
That's a good question. I would say in these cases local vector should match global vector, and local matrix should match global matrix. I believe the scalar types should be derived from the global objects.